pkg/vgmanager/vgmanager_controller.go

```diff
@@ -226,6 +234,71 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request, volumeGr
 	return ctrl.Result{RequeueAfter: requeueAfter}, nil
 }
 
+func (r *VGReconciler) processDelete(ctx context.Context, volumeGroup *lvmv1alpha1.LVMVolumeGroup) error {
+
+	//Read the lvmd config file
+	lvmdConfig, err := loadLVMDConfig()
+	if err != nil {
+		//Failed to read lvmdconfig file. Reconcile again
+		r.Log.Error(err, "failed to read the lvmd config file")
+		return err
+	}
+	if lvmdConfig == nil {
+		r.Log.Info("lvmd config file does not exist")
+		return nil
+	}
+	// avoid having to iterate through device classes multiple times, map from name to config index
+	deviceClassMap := make(map[string]int)
+	for i, deviceClass := range lvmdConfig.DeviceClasses {
+		deviceClassMap[deviceClass.Name] = i
+	}
+	index, found := deviceClassMap[volumeGroup.Name]
+	if !found {
+		// Nothing to do here.
+		r.Log.Info("failed to find volume group in lvmd deviceclasses list", "name", volumeGroup.Name)
+		return nil
+	}
+
+	// Check if volume group exists
+	vg, err := GetVolumeGroup(r.executor, volumeGroup.Name)
+	if err != nil {
+		if err != ErrVolumeGroupNotFound {
+			return fmt.Errorf("failed to get volume group. %v, %v", volumeGroup.Name, err)
+		}
+		return nil
```

Comment:
- why can't directly `return err`?
- if not above, why `return nil` in line 268?

Reply:
> ErrVolumeGroupNotFound , there is nothing more to be done.
- ya, missed this detail